ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -636,4 +621,66 @@ private static boolean isDynPartAbort(Table t, CompactionInfo ci) {
     return t.getPartitionKeys() != null && t.getPartitionKeys().size() > 0
         && ci.partName == null;
   }
+
+  /**
+   * Keep track of the compaction's transaction and its operations.
+   */
+  private class CompactionTxn {
```

Comment:
What if we extend AutoCloseable?
Then we can use the try withresource without finally magic. And close could say - if needed commit.

I think this is the last comment.
How do you like this code? Is it better?